From 46c38b71222b79f711983164a41a4e9d03283016 Mon Sep 17 00:00:00 2001
From: Felix Riemann <friemann@gnome.org>
Date: Wed, 19 Nov 2014 19:20:03 +0100
Subject: [PATCH] Make sure the libpeas-gtk dependency isn't optimized away

Aggressive linkers could drop the dependency which would
cause problems using during runtime.
Based on a patch for the same problem for gedit by Garret Regier.

https://bugzilla.gnome.org/show_bug.cgi?id=739618
---
 src/eog-preferences-dialog.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/eog-preferences-dialog.c b/src/eog-preferences-dialog.c
index d80c8b1..492069f 100644
--- a/src/eog-preferences-dialog.c
+++ b/src/eog-preferences-dialog.c
@@ -144,6 +144,10 @@ eog_preferences_dialog_class_init (EogPreferencesDialogClass *klass)
 {
 	GtkWidgetClass *widget_class = (GtkWidgetClass*) klass;
 
+	/* This should make sure the libpeas-gtk dependency isn't
+	 * dropped by aggressive linkers (#739618) */
+	g_type_ensure (PEAS_GTK_TYPE_PLUGIN_MANAGER);
+
 	gtk_widget_class_set_template_from_resource (widget_class,
 						     "/org/gnome/eog/ui/eog-preferences-dialog.ui");
 	gtk_widget_class_bind_template_child_private (widget_class,
-- 
2.1.3

